src/CauserResolver.php

```diff
@@ -0,0 +1,115 @@
+<?php
+namespace Spatie\Activitylog;
+
+use Closure;
+use Illuminate\Auth\AuthManager;
+use Illuminate\Contracts\Config\Repository;
+use Illuminate\Database\Eloquent\Model;
+use InvalidArgumentException;
+use Spatie\Activitylog\Exceptions\CouldNotLogActivity;
+
+class CauserResolver
+{
+    protected AuthManager $authManager;
+
+    protected string $authDriver;
+
+    /**
+     * User defined model or callback to override default reslover logic
+     */
+    protected Model | Closure | null $override = null;
+
+    public function __construct()
+    {
+        $config = app(Repository::class);
+        $this->authManager = app(AuthManager::class);
+
+        $this->authDriver = $config['activitylog']['default_auth_driver'] ?? $this->authManager->getDefaultDriver();
+    }
+
+    /**
+     * Reslove causer based different arguments
+
+     * @param Model|int|Closure|null $subject
+     * @return null|Model
+     * @throws InvalidArgumentException
+     * @throws CouldNotLogActivity
+     */
+    public function resolve(Model | int | Closure | null $subject = null) : ?Model
```

Comment:
Could we adjust this methods logic to something like the following:

```php
if($this->override !== null) return $this->override;

if($subject === null) return $this->getCauser();

return $this->getCauser($subject);
```

The idea here is:
* always return enforced causer if defined
* return default/current/auth user if `null`/nothing given
* resolve user using the default or given resolver callback